base/ltoutput.dtx

```diff
@@ -2895,17 +2897,29 @@
   \language\document@default@language
 %    \end{macrocode}
 %
-%    The |\catcode`\ = 10| was removed as it was considered useless
-%    (presumably because nothing gets tokenized during shipout).
-%
-%    This was put in as some error produced active spaces in a mark, I
-%    think.
-%
-%    Why was the hyphen reset?
-%
+%    Now a few more resets for things that may have changed during
+%    galley processing. For some active character definitions.
 %    \begin{macrocode}
   \@resetactivechars
 %    \end{macrocode}
+%    Then all kind of catcodes that might got altered by
+%    \texttt{verbatim} or
+%    similar environments. \cs{nfss@catcodes} contains most of what we
+%    need so we use this.
+%  \changes{v1.4o}{2025/02/25}{Reset various catcodes to their default
+%    values in case the page is broken in some unfortunately place
+%    while a special catcode scheme is in force (gh/600)}
+%    \begin{macrocode}
+  \nfss@catcodes
+%    \end{macrocode}
+%    But it makes space a noop and it is missing a few catcodes we
```

Comment:
noop   or   no-op ? 

And:
. . . it misses a few catcodes that also need to be reset.  So we deal with these here: